R/data.table.R

```diff
@@ -1340,6 +1340,10 @@ replace_dot_alias = function(e) {
         setattr(jval, 'sorted', key(x))
       if (any(sapply(jval, is.null))) stop("Internal error: j has created a data.table result containing a NULL column") # nocov
     }
+    
+    # should drop the AsIs class #4326
+    if (inherits(jval, "AsIs")) class(jval) = setdiff(class(jval), "AsIs")
```

Comment:
is `class<-` intentional instead of `setattr`?

Reply:
You are right. This will lead to a copy. I should have used `setattr()`. Fixed.